consensus/consensus_v2.go

```diff
@@ -136,14 +136,19 @@ func (consensus *Consensus) onAnnounce(msg *msg_pb.Message) {
 		return
 	}
 
+	if blockObj.NumberU64() != recvMsg.BlockNum || recvMsg.BlockNum < consensus.blockNum {
+		utils.GetLogger().Warn("blockNum not match", "recvBlockNum", recvMsg.BlockNum, "blockObjNum", blockObj.NumberU64(), "myBlockNum", consensus.blockNum)
+		return
+	}
+
 	// check block data transactions
-	if err := consensus.VerifyHeader(consensus.ChainReader, blockObj.Header(), false); err != nil {
-		utils.GetLogInstance().Warn("onAnnounce block content is not verified successfully", "error", err)
+	if err := consensus.VerifyHeader(consensus.ChainReader, blockObj.Header(), false); err != nil && consensus.mode.Mode() != Syncing {
+		utils.GetLogInstance().Warn("onAnnounce block content is not verified successfully", "error", err, "inChain", consensus.ChainReader.CurrentHeader().Number, "have", blockObj.Header().ParentHash)
 		return
 	}
 	if consensus.BlockVerifier == nil {
 		// do nothing
-	} else if err := consensus.BlockVerifier(&blockObj); err != nil {
+	} else if err := consensus.BlockVerifier(&blockObj); err != nil && consensus.mode.Mode() != Syncing {
```

Comment:
same here. Logic will be cleaner that way.

Reply:
sure